flake8_quotes/__init__.py

```diff
@@ -145,6 +150,11 @@ def parse_options(cls, options):
         if hasattr(options, 'docstring_quotes') and options.docstring_quotes is not None:
             cls.config.update(cls.DOCSTRING_QUOTES[options.docstring_quotes])
 
+        # If allow-escaped specified, add to config
+        if hasattr(options, 'avoid_escape') and options.avoid_escape is not None:
+            avoid_escape = {'true': True, 'false': False}[options.avoid_escape]
```

Comment:
Does the config parser not support booleans? It looks close to `argparse` under the hood (don't have time fully look at the moment). Maybe we should change it over from `type='choice'` to `type=bool`?

https://docs.python.org/2/library/argparse.html#type

Reply:
I can double check, but when I traced the call stack I seemed to get to `optparse` rather than `argparse`, which has pretty restrictive set of types.

https://docs.python.org/3/library/optparse.html#optparse-standard-option-types

Reply:
Ah, I see. It looks like we can prob do this coercion via `callback` support and an inline lambda:

https://docs.python.org/3/library/optparse.html#optparse-option-callbacks

This feels like overkill and I'd be surprised if other `flake8` libraries haven't handled booleans yet. It looks like we can do that via `store_true`/`store_false` and options like `--no-avoid-escape`:

https://docs.python.org/3/library/optparse.html#handling-boolean-flag-options

Going to see how/if that would work in our config file and other `flake8` libraries

Reply:
That was a very annoying rabbit hole. It looks like we are indeed stuck with this strategy. What I learned:

- Some libraries look like they support `--no` based options but they fail to properly support them via `config` (e.g. `flake8-isort`)
- Flake8 requires values for any key in config file to show up in parsed data
    - i.e. `no-avoid-escape` won't show up but `no-avoid-escape = False` would (though it would set `avoid-escape = False` which is very confusing)

Sanest solution to avoid string coercion dancing, use simple boolean flags (e.g. `--avoid-escape`, `--no-avoid-escape`), and stay consistent with other libraries:

- Set up both `--avoid-escape` and `--no-avoid-escape` (with `dest: avoid_escape`) options which use `store_true/false` respectively
    - This allows `avoid-escape = True/False` in config files
    - This allows `flake8 --no-avoid-escape`
    - Reference diff on top of PR: https://github.com/zheller/flake8-quotes/compare/zheller:450806c...zheller:5ccd2f0

Reply:
Request to resolve before landing

Reply:
Noted, and I will try this out when I next get a chance :)

Reply:
Your changes looked good. Have squashed, rebased, and pushed.